service/api/service.go

```diff
@@ -117,6 +120,39 @@ func (s *serviceImpl) ApiV1WorkflowStartPost(ctx context.Context, req iwfidl.Wor
 	}, nil
 }
 
+func (s *serviceImpl) ApiV1WorkflowWaitForStateCompletion(ctx context.Context, req iwfidl.WorkflowWaitForStateCompletionRequest) (wresp *iwfidl.WorkflowWaitForStateCompletionResponse, retError *errors.ErrorAndStatus) {
+	defer func() { log.CapturePanic(recover(), s.logger, &retError) }()
+
+	workflowId := service.IwfSystemConstPrefix + req.WorkflowId + "_" + req.StateExecutionId
+	options := client.StartWorkflowOptions{
+		ID:                       workflowId,
+		TaskQueue:                s.taskQueue,
+		WorkflowExecutionTimeout: time.Duration(*req.WaitTimeSeconds) * time.Second,
+	}
+
+	runId, err := s.client.StartWaitForStateCompletionWorkflow(ctx, options)
+	if err != nil {
+		return nil, s.handleError(err)
+	}
+
+	waitTimeSeconds := 60 // default to 60 seconds
```

Comment:
I think we can just pass req.WaitTimeSeconds into TrimContextByTimeoutWithCappedDDL as a pointer